### PR TITLE
Project system improvement

### PR DESCRIPTION
**Pull request by huoyaoyuan**
Use a single file to manage language and framework version.
Removed some unused fields in project file.

_Any plan to drop VS2015? This allows C# 7, and refining project file to the new format, using .net core SDK and NuGet 4._

### CODE REVIEW

**Comment by peppy** [comment]
2015 can be dropped whenever you'd like.

**Comment by huoyaoyuan** [comment]
Going to add more commits dropping 2015.
This will need tests on non-Windows platforms, to ensure the new version works.
And may notify everyone to install a new SDK if necessary.
_And also update README for the requirements._

**Comment by smoogipoo** [comment]
Is this compatible with the changes in https://github.com/ppy/osu-framework/issues/1098? Can you try merging it in and seeing how many conflicts there are/resolving them?

File: SampleGame/SampleGame.csproj [hunk 1]
```diff
@@ -1,157 +1,24 @@
...
   <ItemGroup>
-    <Reference Include="mscorlib" />
-    <Reference Include="OpenTK, Version=3.0.0.0, Culture=neutral, PublicKeyToken=bad199fe84eb3df4, processorArchitecture=MSIL">
-      <HintPath>$(SolutionDir)\packages\OpenTK.3.0.0-git00009\lib\net20\OpenTK.dll</HintPath>
-      <Private>True</Private>
-    </Reference>
-    <Reference Include="System" />
+    <Content Include="OpenTK.dll.config">
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+    </Content>
```
Comment (smoogipoo) [inline SampleGame/SampleGame.csproj]: *.dll.config files are not supported in .NET Core and we will need per-arch conditional compilation to copy the right assemblies to the output directory. These includes can be removed.

Comment (huoyaoyuan) [reply SampleGame/SampleGame.csproj]: Removing them in the same commit/PR changing the target framework will preserve more compatibility.

Comment (smoogipoo) [reply SampleGame/SampleGame.csproj]: Ah right my bad this is still targetting .net framework, >_<

File: osu.Framework/osu.Framework.csproj [hunk 2]
```diff
@@ -1,633 +1,60 @@
...
-  <!-- To modify your build process, add your task inside one of the targets below and uncomment it.
-       Other similar extension points exist, see Microsoft.Common.targets.
-  <Target Name="BeforeBuild">
-  </Target>
-  <Target Name="AfterBuild">
-  </Target>
-  -->
-  <Import Project="$(SolutionDir)\packages\SQLitePCLRaw.lib.e_sqlite3.v110_xp.1.0.1\build\SQLitePCLRaw.lib.e_sqlite3.v110_xp.targets" Condition="Exists('$(SolutionDir)\packages\SQLitePCLRaw.lib.e_sqlite3.v110_xp.1.0.1\build\SQLitePCLRaw.lib.e_sqlite3.v110_xp.targets')" />
-  <Import Project="$(SolutionDir)\packages\SQLitePCLRaw.lib.e_sqlite3.osx.1.0.1\build\SQLitePCLRaw.lib.e_sqlite3.osx.targets" Condition="Exists('$(SolutionDir)\packages\SQLitePCLRaw.lib.e_sqlite3.osx.1.0.1\build\SQLitePCLRaw.lib.e_sqlite3.osx.targets')" />
-  <Import Project="$(SolutionDir)\packages\SQLitePCLRaw.lib.e_sqlite3.linux.1.0.1\build\SQLitePCLRaw.lib.e_sqlite3.linux.targets" Condition="Exists('$(SolutionDir)\packages\SQLitePCLRaw.lib.e_sqlite3.linux.1.0.1\build\SQLitePCLRaw.lib.e_sqlite3.linux.targets')" />
```
Comment (smoogipoo) [inline osu.Framework/osu.Framework.csproj]: Reason for removing these imports?

Comment (huoyaoyuan) [reply osu.Framework/osu.Framework.csproj]: `build` folder is first class nuget member, and `.targets` in it should be automatically imported.
See https://docs.microsoft.com/en-us/nuget/create-packages/creating-a-package#including-msbuild-props-and-targets-in-a-package

**Comment by smoogipoo** [comment]
LGTM! Happy to finally get this merged in >_<

**Pull request merged**